src/+datools/Model.m

```diff
@@ -1,22 +1,26 @@
 classdef Model < handle
+%MODEL this is the base model class 
+%   Handles all the data pertaining to a given model
+%   User can use OTP model or define their own
+%   User may define their choice of time integration method (or MATLODE)
+%   Model errors can be incorporated using the SynthError
 
     properties
-        ODEModel
-        Solver
-        SynthError
+        ODEModel      % The ODE model from OTP or user defined                  
+        Solver        % Time Integration Scheme
+        SynthError    % Model Error
+        TimeSpan      % Time Span for Propogating the Model
     end
 
     properties (Dependent)
-        State
-        NumVars
-        TimeSpan
-        DistanceFunction
+        State               % Sates of the Model
+        NumVars             % Number of State Variables
+        DistanceFunction    % Distance Function for Localization
     end
 
     methods
         function obj = Model(varargin)
 
-
             p = inputParser;
             addRequired(p, 'ODEModel'); %, @(x) isa(x, 'csl.odetestproblems.Problem'));
             addRequired(p, 'Solver', @(x) nargin(x) == 3);
```

Comment:
This breaks for solvers with 3 args plus varargs, as `nargin` would return -3

Reply:
Yeah, I had went around this by wrapping the solver into a lambda. 

ode45 returns -5 for nargin on my system, by the way.

I am not sure how we can improve this...